packages/jaeger-ui/src/model/ddg/transformDdgData.test.js

```diff
@@ -166,21 +166,43 @@ describe('transform ddg data', () => {
   it('throws an error if a path lacks the focalPayloadElem', () => {
     const { simplePath, noFocalPath, doubleFocalPath, focalPayloadElem } = testResources;
     expect(() =>
-      transformDdgData([simplePath, noFocalPath, doubleFocalPath].map(testResources.wrap), focalPayloadElem)
-    ).toThrowError();
+      transformDdgData(testResources.wrap([simplePath, noFocalPath, doubleFocalPath]), focalPayloadElem)
+    ).toThrowError(/focalNode/);
   });
 
   it('creates equal hashes iff paths are equivalent', () => {
     const { focalPayloadElem, doubleFocalPath, longSimplePath, simplePath, wrap } = testResources;
-    const simpleModel = transformDdgData([simplePath, longSimplePath].map(wrap), focalPayloadElem);
-    const reverseModel = transformDdgData([longSimplePath, simplePath].map(wrap), focalPayloadElem);
+    const simpleModel = transformDdgData(wrap([simplePath, longSimplePath]), focalPayloadElem);
+    const reverseModel = transformDdgData(wrap([longSimplePath, simplePath]), focalPayloadElem);
 
     expect(reverseModel).not.toEqual(simpleModel);
     expect(reverseModel).not.toBe(simpleModel);
     expect(reverseModel.hash).toBe(simpleModel.hash);
 
-    const diffModel = transformDdgData([doubleFocalPath].map(wrap), focalPayloadElem);
+    const diffModel = transformDdgData(wrap([doubleFocalPath]), focalPayloadElem);
 
     expect(diffModel.hash).not.toBe(simpleModel.hash);
   });
+
+  it('adds traceIDs to paths from attributes', () => {
+    const { focalPayloadElem, doubleFocalPath, simplePath, wrap } = testResources;
+    const payload = wrap([simplePath, doubleFocalPath]);
+    payload.dependencies.forEach((dependency, i) => {
+      // eslint-disable-next-line no-param-reassign
+      dependency.attributes = [
+        {
+          key: 'exemplar_trace_id',
+          value: `trace ${i} a`,
+        },
```

Comment:
@yurishkuro I don't recall our previous discussion - but is there a particular reason to prefer this to combining multivalued keys in the backend as so? 
```
{ 
key: 'exemplar_trace_id',
values: {'traceId1', 'traceId2`, ...}
}
```

Reply:
Additionally, is there a reason to prefer the above implementation or the live implementation over:
```
attributes: {
  exemplar_trace_id: ['traceID1', 'traceID2', ...],
  [someOtherKey]: ['value0', 'value1'],
},
```